src/api/coupons/operations/deactivate-coupon.ts

```diff
@@ -1,20 +1,29 @@
-/* eslint-disable @typescript-eslint/no-unsafe-assignment */
 import type {CouponType} from 'src/@types/coupon';
 import {STORAGE} from 'src/config';
 import {getFromDDBB, saveInDDBB} from '../shared/couponsDDBBFunctions';
 
 async function deactivateCoupon(id: string): Promise<void> {
   if (!id || !id.length) throw new TypeError('Item id is not defined');
 
-  const activeCoupons: CouponType[] = getFromDDBB(STORAGE.activeCoupons);
-  const inactiveCoupons: CouponType[] = getFromDDBB(STORAGE.inactiveCoupons);
-  const selectedCoupon: CouponType = activeCoupons.find((coupon: CouponType) => coupon.id === id)!;
+  const activeCoupons = getFromDDBB<CouponType[]>(STORAGE.activeCoupons);
+  const inactiveCoupons = getFromDDBB<CouponType[]>(STORAGE.inactiveCoupons);
+  if (!activeCoupons) {
+    throw new Error('Item is not active');
```

Comment:
Quizás en este caso el error podría ser diferente al caso de que no haya un cupón. Algo parecido a la sugerencia
```suggestion
    throw new Error('No items active');
```